app/js/PersistorManager.js

```diff
@@ -1,37 +1,19 @@
-// TODO: This file was created by bulk-decaffeinate.
-// Sanity-check the conversion and remove this comment.
-/*
- * decaffeinate suggestions:
- * DS103: Rewrite code to no longer use __guard__
- * DS205: Consider reworking code to avoid use of IIFEs
- * DS207: Consider shorter variations of null checks
- * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
- */
 const settings = require('settings-sharelatex')
 const logger = require('logger-sharelatex')
 
-// assume s3 if none specified
-__guard__(
-  settings != null ? settings.filestore : undefined,
-  x => x.backend || (settings.filestore.backend = 's3')
-)
+module.exports = (function() {
```

Comment:
We could drop the wrapper function and help IDEs with duck-typing :)

```js
logger.log(
  {
    backend: settings.filestore.backend
  },
  'Loading backend'
)

if (!settings.filestore.backend) {
  throw new Error('no backend specified - config incomplete')
}

switch (settings.filestore.backend) {
  case 'aws-sdk':
    return module.exports = require('./AWSSDKPersistorManager')
  case 's3':
    return module.exports = require('./S3PersistorManager')
  case 'fs':
    return module.exports = require('./FSPersistorManager')
  default:
    throw new Error(
      `unknown filestore backend: ${settings.filestore.backend}`
    )
}
```

Reply:
Good call. It's better not to use `return`s at the top-level (at least, `eslint` doesn't like it) but other than that it's a good change. :+1: